From cd369f020012e7dd776e4ff3e1a083d96aa55be2 Mon Sep 17 00:00:00 2001
From: Eric Kilmer <eric.d.kilmer@gmail.com>
Date: Sat, 8 Feb 2025 17:59:57 -0500
Subject: [PATCH 5/9] decompiler: Fix strict weak ordering TypePartialEnum

This fixes Windows Debug error encountered in testing where it was
complaining about lack of strict weak ordering.
---
 Ghidra/Features/Decompiler/src/decompile/cpp/type.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Ghidra/Features/Decompiler/src/decompile/cpp/type.cc b/Ghidra/Features/Decompiler/src/decompile/cpp/type.cc
index 6b226036cb..2ff7c69090 100644
--- a/Ghidra/Features/Decompiler/src/decompile/cpp/type.cc
+++ b/Ghidra/Features/Decompiler/src/decompile/cpp/type.cc
@@ -2830,6 +2830,7 @@ int4 TypePartialEnum::compareDependency(const Datatype &op) const
 
 {
   if (submeta != op.getSubMeta()) return (submeta < op.getSubMeta()) ? -1 : 1;
+  if (parent == &op) return 1;	// op is our TypeEnum
   TypePartialEnum *tp = (TypePartialEnum *) &op;	// Both must be partial
   if (parent != tp->parent) return (parent < tp->parent) ? -1 : 1;	// Compare absolute pointers
   if (offset != tp->offset) return (offset < tp->offset) ? -1 : 1;
-- 
2.54.0 (Apple Git-157)

